subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/Codecs.kt

```diff
@@ -232,17 +231,19 @@ class Codecs(
     }
 
     private
-    fun Bindings.completeWithReentrantBeanCodec() = append {
-        // This protects the BeanCodec against StackOverflowErrors but
+    fun Bindings.completeWithStatefulCodecs() = append {
+        bind(JavaObjectSerializationCodec(javaSerializationEncodingLookup))
```

Comment:
🤔 I'm wondering if it wouldn't be better for loading performance (more sharing of these reflection lookup results) to synchronize access to `readResolveMethod` and `readObjectHierarchy` instead.

I think we should benchmark one option against the other using Gradle profiler. WDYT?

Reply:
See https://github.com/gradle/gradle/issues/26400#issuecomment-1729275552

Reply:
@bamboo So we are ok to land it, right?